File: FateGrandAutomata/Imaging/DisposableMat.cs
using System;
using Org.Opencv.Core;

namespace FateGrandAutomata
{
    class DisposableMat : IDisposable
    {
        public Mat Mat { get; private set; }

        public DisposableMat() : this(new Mat()) { }
        
        public DisposableMat(Mat Mat)
        {
            this.Mat = Mat;
        }

        public void Dispose()
        {
            Mat.Release();
            Mat = null;
        }
    }
}

File: app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DisposableMat.kt
package com.mathewsachin.fategrandautomata.imaging

import org.opencv.core.Mat

class DisposableMat(var mat: Mat? = Mat()) : AutoCloseable {
    init {
        require(mat != null) { "Mat should not be null" }
    }

    constructor() : this(Mat())

    override fun close() {
        mat?.release()
        mat = null
    }
}